tdms/src/interpolate_Efield.cpp

```diff
@@ -2,56 +2,56 @@
 #include "interpolation_methods.h"
 
 
-void interpolateTimeDomainEx(double ***Exy, double ***Exz, int i, int j, int k, int I, double *Ex) {
+void interpolateTimeDomainEx(double ***Exy, double ***Exz, int i, int j, int k, int nI, double *Ex) {
 
     // determine the interpolation scheme to use
-    const interpScheme &scheme = best_interp_scheme(I, i);
+    const interpScheme &scheme = best_interp_scheme(nI, i);
     // prepare input data - if using a cubic scheme we have reserved more memory than necessary but nevermind
     double interp_data[8];
 
     // now fill the interpolation data
-    // i + scheme.index-1 is the index of the Yee cell that plays the role of v0 in the interpolation
-    // note that scheme.index-1 appears here because Yee cells associate field values "to the right" of their centre (IE, the field component is at a position with a higher coordinate value than the Yee cell centre)
+    // i - (scheme.index+1) is the index of the Yee cell that plays the role of v0 in the interpolation
+    // note that scheme.index+1 apears here because Yee cells associate field values "to the right" of their centre (IE, the field component is at a position with a higher coordinate value than the Yee cell centre)
     for(int ind=scheme.first_nonzero_coeff; ind<=scheme.last_nonzero_coeff; ind++) {
-        interp_data[ind] = Exy[k][j][i - (scheme.index-1) + ind] + Exz[k][j][i - (scheme.index-1) + ind];
+        interp_data[ind] = Exy[k][j][i - (scheme.index + 1) + ind] + Exz[k][j][i - (scheme.index + 1) + ind];
     }
 
     // now run the interpolation scheme and place the result into the output
     *Ex = scheme.interpolate(interp_data);
 }
 
-void interpolateTimeDomainEy(double ***Eyx, double ***Eyz, int i, int j, int k, int J, double *Ey)
+void interpolateTimeDomainEy(double ***Eyx, double ***Eyz, int i, int j, int k, int nJ, double *Ey)
 {
     // determine the interpolation scheme to use
-    const interpScheme &scheme = best_interp_scheme(J, j);
+    const interpScheme &scheme = best_interp_scheme(nJ, j);
     // prepare input data - if using a cubic scheme we have reserved more memory than necessary but nevermind
     double interp_data[8];
 
     // now fill the interpolation data
-    // j + scheme.index-1 is the index of the Yee cell that plays the role of v0 in the interpolation
-    // note that scheme.index-1 appears here because Yee cells associate field values "to the right" of their centre (IE, the field component is at a position with a higher coordinate value than the Yee cell centre)
+    // j - (scheme.index+1) is the index of the Yee cell that plays the role of v0 in the interpolation
+    // note that scheme.index+1 apears here because Yee cells associate field values "to the right" of their centre (IE, the field component is at a position with a higher coordinate value than the Yee cell centre)
```

Comment:
The `index` field [is documented with a note](https://github.com/UCL/TDMS/blob/80c1b1e8a74952718bb17472a4397c7f9a5b3db0/tdms/include/interpolation_methods.h#L112-L117) concerning why this `+1` later appears. `index` is definitely not the most informative name choice, but equally I couldn't think of anything succinct to name this variable. Using your suggestion I'll change its name to `number_of_datapoints_to_left` or something, which better describes it's new role. `BAND_LIMITED_CELL_ZERO` will have a rather value for this variable, but otherwise it'll probably make things clearer.

My original idea was that `index` matches the `interpolation position` labeling that is set up and described [here](https://github.com/UCL/TDMS/blob/80c1b1e8a74952718bb17472a4397c7f9a5b3db0/tdms/include/interpolation_methods.h#L64-L68). Then, the `+1` seems fairly natural to me since the Yee cells always associate field components "to the right" of their centres (IE, the centre of a Yee cell is never has a coordinate greater than those of the position of any field components associated to that cell). A similar logic also holds for the cubic schemes. It also means we can avoid adjustments to the code when we want to interpolate "to the left" of cell 0 (`BAND_LIMITED_CELL_ZERO`).